### PR TITLE
ONYX-11829: Refactor Authenticator Parameters

### PR DESCRIPTION
**Pull request by tzheleznyak**
### Whats in the PR

* Rename to JwtAuthenticatorInput
* Remove attr accessors
* Create only after validate and decode
* Add deligators where relevant for JwtAuthenticatorInput and AuthenticatorInput

Commit 1 - Code Changes
Commit 2 - Rspec changes

### What ticket does this PR close?
ONYX-11829

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [x] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb [hunk 1]
```diff
@@ -51,18 +54,18 @@ def identity_path_resource_exists?
...
         end
 
         def identity_path_secret_value
           @identity_path_secret_value ||= @fetch_authenticator_secrets.call(
-            conjur_account: @authentication_parameters.account,
-            authenticator_name: @authentication_parameters.authenticator_name,
-            service_id: @authentication_parameters.service_id,
+            conjur_account: account,
+            authenticator_name: authenticator_name,
+            service_id: ervice_id,
```
Comment (benoaviram) [inline app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb:68]: ?

Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb:68]: Fixed

File: app/domain/authentication/authn_jwt/jwt_authenticator_input.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Authentication
+  module AuthnJwt
+    # Data class to store data regarding jwt token that is needed during the jwt authentication process
+    # :reek:TooManyInstanceVariables and :reek:TooManyParameters
+    class JWTAuthenticatorInput
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/jwt_authenticator_input.rb:5]: do we have a plan to unify between this class and `AuthenticatorInput`? if so, can we please create a JIRA issue for RnD Boost and link to it here?

File: app/domain/authentication/authn_jwt/validate_status.rb [hunk 3]
```diff
@@ -94,44 +94,48 @@ def validate_webservice_is_whitelisted
...
-            service_id: service_id,
-            account: account,
-            username: username,
-            client_ip: client_ip,
-            credentials: nil,
-            request: nil
-          ),
-          jwt_token: nil
+      def jwt_authenticator_input
+        @jwt_authenticator_input ||= Authentication::AuthnJwt::JWTAuthenticatorInput.new(
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/validate_status.rb:124]: should this be a dependency?

File: app/domain/authentication/authn_jwt/validate_status.rb [hunk 4]
```diff
@@ -94,44 +94,48 @@ def validate_webservice_is_whitelisted
...
+          authenticator_input: authenticator_input,
+          decoded_token: nil
+        )
+      end
+
+      def authenticator_input
+        @authenticator_input ||= Authentication::AuthenticatorInput.new(
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/validate_status.rb:131]: should this be a dependency?

File: app/domain/authentication/authn_jwt/jwt_authenticator_input.rb (hunk 2 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/jwt_authenticator_input.rb:5]: But this class is good for jwt authenticators. Other authenticatiors don't need this . So i think they can remain seperatly

**Review by orenbm — DISMISSED** [review]

File: app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb (hunk 1 above)
Comment (nessiLahav) [inline app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb]: `ervice_id` typo

Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb]: Already fixed

**Review by nessiLahav — APPROVED** [review]
LGTM

**Pull request merged**